### ISSUE TITLE
Incorrect v2 storage account kind returned from list / get operation

### ISSUE THREAD

**Issue by MichaelBurgess** [Azure/azure-sdk-for-node#2396]
Hi,

I raised an issue with Azure support, as I was seeing that the Storage Accounts List / Get operations started returning the wrong kind for a v2 storage account.

If I perform a list operation in the [resources explorer](https://resources.azure.com/subscriptions/<subscriptionId>/resourceGroups/<resourceGroupName>/providers/Microsoft.Storage/storageAccounts), or via the [Node.js SDK](https://docs.microsoft.com/en-gb/javascript/api/azure-arm-storage/storageaccounts?view=azure-node-2.2.0), my v2 storage accounts are coming in as kind `Storage`, rather than `StorageV2`

Azure support have confirmed that the API version we need to be using in order to fix this is `2017-10-01` and not `2017-06-01` and I have confirmed that this works by making a direct `sendRequest` via the Node.js SDK to the REST API with the newer API version.

We're using `"azure": "2.2.1-preview"` and `azure-arm-storage` seems to be `3.1.0` by default.

I can see that `azure-arm-storage 3.2.0` has been published and this contains the required API version (`2017-10-01`), however when I upgrade to this version I am still seeing a v2 storage kind coming back as `Storage`.

I've also tried catching the raw response in the callback and parsing the body, but the v2 storage account kind still shows as `Storage`.

Do I need to change anything else for this to work? In the meanwhile we can use a direct REST API call with `sendRequest`.

Thanks,
Mike.

**Comment by amarzavery** [comment]
Well I tried to regenerate the sdk locally from the 2017-10-01 api-version of the swagger spec and did not see any changes in the generated sdk. Which means that the generated sdk is the latest.

Can you please provide us the `request` and `response (body and headers)` objects for both the approaches:
- when you send via sendRequest()
- when you use the sdk method.

That should tell us what is different.

Adding @JasonYang-MSFT from the storage team. Jason do you have any inputs on the issue reported by Mike?

**Comment by JasonYang-MSFT** [comment]
Acked.

**Comment by MichaelBurgess** [comment]
@amarzavery sorry for the delay in replying - will get the required info for you as soon as I can.

**Comment by MichaelBurgess** [comment]
Hi @amarzavery, I just got round to looking into this again.

This is user error sorry which I discovered whilst getting you the request headers - although I had updated the `azure-arm-storage` dependency, I hadn't required it in my script and was still using `azure.createStorageManagementClient` rather than the `azure-arm-storage` class. As a result the header was showing `3.1.0` still for the storage client version.
